clients/ereporter-client/src/lib.rs

```diff
@@ -0,0 +1,26 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+progenitor::generate_api!(
+    spec = "../../openapi/ereporter/ereporter-latest.json",
+    inner_type = slog::Logger,
+    derives = [schemars::JsonSchema, Clone, Eq, PartialEq],
+    pre_hook = (|log: &slog::Logger, request: &reqwest::Request| {
+        slog::debug!(log, "client request";
+            "method" => %request.method(),
+            "uri" => %request.url(),
+            "body" => ?&request.body(),
+        );
+    }),
+    post_hook = (|log: &slog::Logger, result: &Result<_, _>| {
+        slog::debug!(log, "client response"; "result" => ?result);
+    }),
+    replace = {
+        Ena = ereport_types::Ena,
+        Ereport = ereport_types::Ereport,
+        Event = ereport_types::Event,
+        ReportKind = ereport_types::ReportKind,
```

Comment:
I'm not sure you need this.  They don't seem to have functionality defined on them?

(I am generally kind of phobic about `replace`.  It makes sense when there's non-trivial functionality on a type, but it feels risky if the types don't wind up matching.)

Reply:
Yeah, upon further consideration this is not actually accomplishing anything of value, I'll remove it.

Reply:
Mainly, I thought it would be nice for the client (Nexus) to also depend on the types crate and use them for stuff like database operations without having to do a ton of type conversions, but I get the concern...